### PR TITLE
DM-12851: Determine whether dataset framework can be made generic

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR updates the instructions to clarify what is and is not required (very little) when filling in the `raw` and `calib` directories. The out-of-date description of `templates` is also fixed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,12 +8,12 @@ This repo is designed to be used as a template for developing new data sets for
...
 
 Relevant Files and Directories
------
+------------------------------
 path                  | description
 :---------------------|:-----------------------------
-`raw`                 | To be populated with raw data. Currently contains a single small fits file (taken from `obs_test`) to test `git-lfs` functionality.
-`calib`               | To be populated with master calibs. Currently empty.
-`templates`           | To be populated with calibrated images intended to be used as templates. Currently empty.
+`raw`                 | To be populated with raw data. May be organized however is convenient. Currently contains a single small fits file (taken from `obs_test`) to test `git-lfs` functionality.
```
Comment (mrawls) [inline README.md]: Out of context, it's not clear what "May be organized however is convenient" for `raw` means. I suppose it's in contrast to the specification below that `templates` must be organized as a filesystem-based Butler repo? It might be clearer to say that subdirectories are OK and that any images dumped in here will be ingested (assuming that's what you mean!). Same goes for `calib` description below.

File: README.md [hunk 2]
```diff
@@ -8,12 +8,12 @@ This repo is designed to be used as a template for developing new data sets for
 It is currently set up for using `obs_test`.
```
Comment (mrawls) [inline README.md]: Please add a link to and a super brief description of what the `obs_test` package is. Does this mean the dataset template is not intended for use with other obs packages (yet)? Sorry if this is a bit out of scope, this is my first look at this lovely new package 👍

Comment (kfindeisen) [reply README.md]: This package predates `ap_verify_hits2015`, I'm surprised you didn't know about it. :confused: 

I think the meaning is that the template is configured to use `obs_test` instead of a "real" package like `obs_decam`. Is there a concise way to say that?

File: README.md (hunk 1 above)
Comment (kfindeisen) [reply README.md]: I want to keep it brief; how about "Subdirectories are allowed but not required."?

File: README.md (hunk 2 above)
Comment (mrawls) [reply README.md]: My mistake, I probably knew about it and forgot... how about "this template dataset is configured for use with `obs_test` (include link), which is itself a framework(template?) for real obs_ packages in the LSST Stack." Perhaps not as concise as you'd like.

**Review by mrawls — APPROVED** [review]
Looks good!
